src/collision/collider/parry/mod.rs

```diff
@@ -868,7 +868,7 @@ impl Collider {
             "Each row in `heights` must have the same amount of points"
         );
 
-        let heights = nalgebra::DMatrix::from_vec(row_count, column_count, data);
+        let heights = nalgebra::DMatrix::from_vec(column_count, row_count, data);
```

Comment:
Do not pass `column_count` to the `nrows` argument.

DMatrix::from_vec's first argument is `nrows`, and second is `ncol`; the bug is in the calculation of column and row count.
Instead of switching the order of the arguments, the two variables' assignments should be switched.